airflow/contrib/utils/field_validator.py

```diff
@@ -0,0 +1,417 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Validator for body fields sent via GCP API.
```

Comment:
Now that this is in contrib/utils, we should probably remove GCP references

Reply:
@TrevorEdwards  I think maybe instead of making it super generic, we could leave it as gcp_body_field_validator.py and name the class GcpBodyFieldValidator ? 

While it is pretty generic indeed, it has certain GCP-ness to it (api_version for one, approach to union fields is also fairly GCP-specific and apparently similar for different GCP services). I'd say at this stage we should simply focus to make it reusable across GCP operators. Later on if someone wants to use it for other operators, they could make a change to rename it and make it even more generic (or even we can do it at some point in time). This might even result in extending its functionality for other non-GCP operators.

I am always in favour in making things reusable and generic exactly at the moment when it is actually needed, my experience is that first you need to make classes usable before they can be re-usable ;).

Reply:
Renaming to gcp* sounds good. Just want to make sure it is clear to Airflow whether we are contributing a general-purpose tool or gcp-specific one.

Reply:
Done in 0be1082eadfe08c096c3d2b7677e192a006f81eb